client/payment-details/payment-method/index.js

```diff
@@ -11,25 +11,33 @@ import { Card, CardBody, CardHeader } from '@wordpress/components';
  */
 import Loadable from 'components/loadable';
 import CardDetails from './card';
+import CardPresentDetails from './card-present';
 import GiropayDetails from './giropay';
 import SepaDetails from './sepa';
 import SofortDetails from './sofort';
 
 const PaymentDetailsPaymentMethod = ( { charge = {}, isLoading } ) => {
-	let PaymentMethodDetails = CardDetails;
+	const DetailComponents = {
+		card: CardDetails,
+		card_present: CardPresentDetails, // eslint-disable-line camelcase
+		giropay: GiropayDetails,
+		sepa_debit: SepaDetails, // eslint-disable-line camelcase
+		sofort: SofortDetails,
+	};
```

Comment:
🔍 Few nitpicking things which would make it polished:
1. Since the `type: component` pairs are static moving it to the module level constant (out of PaymentDetailsPaymentMethod function) makes sense, it's normal practice in JS.
2. First upper case letter is used for classes and components, other variables should be named with lower case, so `detailsComponentMappings` would be better name IMO.

Reply:
Changed in https://github.com/Automattic/woocommerce-payments/pull/1840/commits/7b0ff25db73bcf423b01448f7deb6f74477a9740